velox/serializers/PrestoSerializer.cpp

```diff
@@ -587,6 +591,9 @@ void read<StringView>(
   const int32_t size = source->read<int32_t>();
   const int32_t numNewValues = sizeWithIncomingNulls(size, numIncomingNulls);
 
+  if (!result->isFlatEncoding()) {
```

Comment:
nit: would `BaseVector::flattenVector` work instead?

Reply:
Good call, that would be a more succinct way to write this.

Reply:
Although I don't want to do this for the complex types, both ensureWritable and flattenVector recursively flatten children.  I don't want to flatten the children unless I have to.